kernels/volk/volk_64f_x2_dot_prod_64f.h

```diff
@@ -0,0 +1,842 @@
+{
+    unsigned int number;
+    const unsigned int eighthPoints = num_points / 8;
+
+    const double* aPtr = input;
+    const double* bPtr = taps;
+
+    __m512d dotProdVal = _mm512_setzero_pd();
+    __m512d aVal1, bVal1;
+
+    for (number = 0; number < eighthPoints; number++) {
+
+        aVal1 = _mm512_loadu_pd(aPtr);
+        bVal1 = _mm512_loadu_pd(bPtr);
+        aPtr += 8;
+        bPtr += 8;
+
+        dotProdVal = _mm512_fmadd_pd(aVal1, bVal1, dotProdVal);
+    }
+
+    __VOLK_ATTR_ALIGNED(64) double dotProductVector[8];
+    _mm512_storeu_pd(dotProductVector,
+                     dotProdVal); // Store the results back into the dot product vector
+
+    double dotProduct = dotProductVector[0] + dotProductVector[1] + dotProductVector[2] +
+                       dotProductVector[3] + dotProductVector[4] + dotProductVector[5] +
+                       dotProductVector[6] + dotProductVector[7];
+
+    for (number = eighthPoints * 8; number < num_points; number++) {
+        dotProduct += ((*aPtr++) * (*bPtr++));
+    }
+
+    *result = dotProduct;
+}
+#endif /* LV_HAVE_AVX512F */
+
+#endif /*INCLUDED_volk_64f_x2_dot_prod_64f_u_H*/
+
+#ifndef INCLUDED_volk_64f_x2_dot_prod_64f_a_H
+#define INCLUDED_volk_64f_x2_dot_prod_64f_a_H
+
+#include <stdio.h>
+#include <volk/volk_common.h>
+
+
+#ifdef LV_HAVE_GENERIC
+
+
+static inline void volk_64f_x2_dot_prod_64f_a_generic(double* result,
+                                                      const double* input,
+                                                      const double* taps,
+                                                      unsigned int num_points)
+{
+
+    double dotProduct = 0;
+    const double* aPtr = input;
+    const double* bPtr = taps;
+    unsigned int number = 0;
+
+    for (number = 0; number < num_points; number++) {
+        dotProduct += ((*aPtr++) * (*bPtr++));
+    }
+
+    *result = dotProduct;
+}
+
+#endif /*LV_HAVE_GENERIC*/
+
+
+#ifdef LV_HAVE_SSE
+
+
+static inline void volk_64f_x2_dot_prod_64f_a_sse(double* result,
+                                                  const double* input,
+                                                  const double* taps,
+                                                  unsigned int num_points)
+{
+
+    unsigned int number = 0;
+    const unsigned int eighthPoints = num_points / 8;
+
+    double dotProduct = 0;
+    const double* aPtr = input;
+    const double* bPtr = taps;
+
+    __m128d a0Val, a1Val, a2Val, a3Val;
+    __m128d b0Val, b1Val, b2Val, b3Val;
+    __m128d c0Val, c1Val, c2Val, c3Val;
+
+    __m128d dotProdVal0 = _mm_setzero_pd();
+    __m128d dotProdVal1 = _mm_setzero_pd();
+    __m128d dotProdVal2 = _mm_setzero_pd();
+    __m128d dotProdVal3 = _mm_setzero_pd();
+
+    for (; number < eighthPoints; number++) {
+
+        a0Val = _mm_load_pd(aPtr);
+        a1Val = _mm_load_pd(aPtr + 2);
+        a2Val = _mm_load_pd(aPtr + 4);
+        a3Val = _mm_load_pd(aPtr + 6);
```

Comment:
This might actually be a source for slow results. Compilers are incredibly smart nowadays. This kind of manual "loop unrolling" might actually block some compiler optimizations.
You might want to start with godbolt.com and inspect the results for the generic kernel in case you compile for a specific SIMD extension. I'm aware that it might not be trivial to find the optimized assembly code in the output. Still, it is a possible starting point.